### PR TITLE
port lighthouse real

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
bring lighthouse back with some major changes:
- redid all of epi
- replaced meme ai room with an actual theatre for doing plays real
- logi reworked, cargo shuttle and not telepad!!!!
- bringing it up to speed in most places its old
- cryooooo
- ai core with airlocks and disposals entrance
- atmos has teg
- armoury has mines so its not completely piss easy to break into it
- replaced broadway med outpost with psychologist office
- random stuff

still needs more test

# Media
station map

render
Lighthouse-0

**Changelog**
:cl:
- add: Lighthouse is BACK!

### CODE REVIEW

**Comment by deltanedas** [comment]
@IamVelcroboy if you have any big concerns rn id like to hear them, before anything else is finalized

**Comment by IamVelcroboy** [comment]
No concerns until I get a chance to look it over. You'll also need to add it to the map pool located in the prototype folder and the [postmapinittest](https://github.com/DeltaV-Station/Delta-v/blob/master/Content.IntegrationTests/Tests/PostMapInitTest.cs) before it's merged.

Also if you could get a render that would be nice. If not, I'll try to get one tomorrow.

**Comment by deltanedas** [comment]
ill try render it when im awake

**Comment by IamVelcroboy** [comment]
No rush. I'm off to bed soon too. I was told you are not on Discord, so, here is the list of changes we've made since the Rebase. It looks like you got most of them already. Just wanted you to have it if it helps.
- Add the glass box in captain's quarter/office. Do not map the closet containing the antique laser.
- 1 prisoner cryo bed in Perma 
- 1-2 standard cryobeds in medical(optional for smaller maps). 
- Max 4 cryobeds in a public cryo room on most maps. Max 6 on very large maps. These are where latejoin go.

(IMPORTANT-Rotation of cryobeds matters. Players emerge from direction rotated. So, don't send players into walls.)

- Map the spare ID cabinet in bridge or vault
- APCs/Substations/SMES all need to be named now. They show up on the Power Monitoring Console.
- All heads have specific dressers now. These need to be mapped in their rooms as they contain all the drip.
- Warp point beacons were replaced by physical entities that need to be remapped. 
- There is a thief target for Tropico the pet. We are giving this to the detective not atmos.
- Map more "screens" around public halls. They now broadcast command announcments.
- Map the metempsychotic machine in place of the cloner.
- Add Detective room/spawner/closet/update proto
- Make sure Mantis has correct airlock and remove forensic items
- Combine salvage and cargo, add LO Office/room
- Add Salvage shuttle/cargo shuttle
- Swap out QM stuff for logistic tech stuff
- Move robotics, circuit imprinter, and RE machine to Epi.
- Add ID terminals to every Head office
- Add paramedic room(optional)
- Add intern roles(Service worker, Tech Ass, Medical, Research), update proto
- Add borg chargers throughout station
- Update appropriate external airlocks w/ easy-pry
- Link external airlocks
- Make sure head lockers contain hardsuits or map a hardsuit storage
- Add TEG or Singularity Gen(all maps should have 2 sources of power not including solar.)
- Do not map Tesla
- Add Cryogenics in medical(optional but recommended)
- Generators require fuel now. Map a few jr. pacs in maints
- Add ninja bomb targets(not in Epistemics, arrivals, or evac)
- Add grills in kitchens
- Update Prototype file with Job ID Name changes
- Bomb testing room w/ test bombs in sec(optional)
- Hologram for command(optional)
- Add Chemists role/airlocks
- Update O2/N2 gas miners appropriately
- Remove plasma miners. Give atmos a caniser or 2
- Chem labs need condensors, electrolysis, and centrifuge machines
- Add the vault freezer inthe vault
- Remove any Vectors from the map
- Epi should have protolathe and circuit board printer
- Engi should only have autolate  
- Add Corpsman(optional but recommended)
- Jani/Psych/Court signs
- Add CuraDrobe in library
- Ideally every map has a reporter and psychologist(Optional but heavily desired)
- Lawyer has their own access now. Make sure they can get into sec general area but not open cells.
- Do not map the Throngler

**Comment by stellar-novas** [comment]
I tried doing it, got
`[ERRO] map: Error in map file: found local entity UID '0' which does not exist. Exception:`

**Comment by IamVelcroboy** [comment]
> I tried doing it, got `[ERRO] map: Error in map file: found local entity UID '0' which does not exist. Exception:`

Looks like there's a borked toy sword and the cargo console is still trying to link to the telepad. Easy to fix. Run the tidy tool if you can. Otherwise I can do it tomorrow.

**Comment by DangerRevolution** [comment]
My discord isn't loading so I'm just gonna leave it here.

I know it's a case of "you snooze, you lose" but @IamVelcroboy might be worth holding a little as last I checked, RK and ODJ are finishing up a compete remaster of Lighthouse. Might be worth comparing 1:1.

RK did say they were burnt out recently and weren't working on it, but then came back to working on it??


TLDR:
I don't know what progress they've made, maybe it's still unfinished, just wanted to make a note of it.

**Comment by deltanedas** [comment]
@OldDanceJacket are you finishing a port if so ill close

and ill clean up the evil uids now

**Comment by DangerRevolution** [comment]
> @OldDanceJacket are you finishing a port if so ill close
> 
> and ill clean up the evil uids now

RK was the main one working on it and I've asked them abt it, I'll let you know :O

**Comment by deltanedas** [comment]
it has no bomb testing room but idk where to put it

**Comment by stellar-novas** [comment]
Assuming you mean space, it could be separate from sec proper like on Submarine.

Could also not include it, it *is* an optional thing.

**Comment by DangerRevolution** [comment]
> it has no bomb testing room but idk where to put it

some maps which don't have a dedicated room just stick it in the armoury and let Sec figure it out

**Comment by stellar-novas** [comment]
>add techie spawns
it's called a tech ass smh

**Comment by IamVelcroboy** [comment]
Got an update from ODJ for ya 
image

**Comment by deltanedas** [comment]
door naming ops i thought it was bad, will have to do a pass of naming everything :trollface:

**Comment by OldDanceJacket** [comment]
> door naming ops i thought it was bad, will have to do a pass of naming everything :trollface:

Naming doors is fairly important for Lighthouse, players have cited it as a very good feature of the map that they did use.

**Comment by Bonktrauma** [comment]
> > door naming ops i thought it was bad, will have to do a pass of naming everything :trollface:
> 
> Naming doors is fairly important for Lighthouse, players have cited it as a very good feature of the map that they did use.

Not to you jacket,
Also, why did you choose to make the top outpost into psych rather then bottom? The top was actually used unlike the bottom

**Comment by deltanedas** [comment]
the top was used by tiders that break in and cook meth

the engi minimed is used reasonably often when engi gets bombed or whatever (its also near epi and evac which are also more heavily injured than, laser tag)

**Comment by deltanedas** [comment]
game tweaking when i try to connect to server and do stuff :trollface:

**Comment by DebugOk** [comment]
> game tweaking when i try to connect to server and do stuff :trollface:

You've pissed off the networking gods, its over

**Comment by IamVelcroboy** [comment]
Just letting you know I didn't forget about this. Trying to find time to do a proper review. Should be soon™.

**Comment by IamVelcroboy** [comment]
- Missing spawnpoint for Research Assistant 
- I'm not going to force it but we have the better looking asteroid rocks and 
- non-mineable mountain rock can be used to prevent spacing depts
- Maint airlock between chapel/library makes no sense
- lightpost on left side of chapel courtyard is unpowered 
- Maybe put small maint lights in confession booth
- overlapping carpets(2 dif colors) in chaplain's room
- 1 of the lights linked to the switch in the chapel is not linked
- Kinda feel like chaplain should be able to access crematory from room
- Various fire alarms not linked to anything. And spaces not effectively controlled by alarms/sensors
- Re-route scrubber pipe between chapel and library to not pass through so many walls
- Move lights off windows in robo lab
- Robo lab has 2 cameras, remove the one on the shutters, move the other off the window
- Disposal unit in Epi not connected and pipes deadend
- floating cameras/cameras on windows
- Conveyor belts and lights in cargo dock not powered
- need tiny fans under cargo dock airlocks
- LOTS of lights need to be culled in the halls 
- Remove obsolete cargo shuttle console and salvage shuttle console 
- More lights on windows in cargo
- "Side Dock" needs tiny fans
- Cables in conference room not attached to any APC
- Tile missing under medical exit airlock(cloning room)
- Botany APC not powered or attached to LV
- Kitchen APC not powered or attached to LV
- Psych office is kinda lacking for light but they do have a lamp so....
- CMO needs ID console
- Jani room not connected to APC
- Some surveillance cameras not capitalized(most others are)

**Comment by IamVelcroboy** [comment]
Example of too much lighting 
Screenshot 2024-03-20 123221

**Comment by deltanedas** [comment]
pretty sure thats from before big light prune

**Comment by IamVelcroboy** [comment]
Just checking, did the commits above address the list of issues? Is it ready to be reviewed again? Or do you have more to do yet?

**Comment by deltanedas** [comment]
everything should be done, besides warp point stuff

**Comment by IamVelcroboy** [comment]
> everything should be done, besides warp point stuff

Great! Here's how we are mapping our WPs now
```
Map the beacons for what you want to show on the map.

Map Warp Points and/or Bomb Targets where you want Ghost Warp to show up in the menu.

If you want to map a Warp Point for a dept but want to put the Bomb Target in a different place or more than one, then remove the WP comp from it.
```
I'll try to get the re-reviewed early next week and possibly added to the pool again. Thanks!

**Comment by IamVelcroboy** [comment]
So, I still think there are a few too many lights in some places but I'm going to merge anyway. Maybe I'll do a pass later when I have some time and adjust those spots.

**Comment by UnicornOnLSD** [comment]
~ANGLE~ LIGHTHOUSE IS REAL :o
